src/Database/Expression/QueryExpression.php

```diff
@@ -366,6 +366,25 @@ public function notIn($field, $values, $type = null)
         return $this->add(new ComparisonExpression($field, $values, $type, 'NOT IN'));
     }
 
+    /**
+     * Adds a new condition to the expression object in the form
+     * "(field NOT IN (value1, value2) OR field IS NULL".
+     *
+     * @param string|\Cake\Database\ExpressionInterface $field Database field to be compared against value
+     * @param string|array|\Cake\Database\ExpressionInterface $values the value to be bound to $field for comparison
+     * @param string|null $type the type name for $value as configured using the Type map.
+     * @return $this
+     */
+    public function notInOrNull($field, $values, $type = null)
```

Comment:
```suggestion
    public function notInOrNull($field, $values, ?string $type = null)
```

Reply:
This is copied from `notIn()` above so it's the same.

Reply:
Sure, but here you do not have to follow BC issue of the signature.